packages/pages/src/common/src/function/types.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * Defines the functions and fields that are available to the serverless function.
+ * @public
+ */
+export interface FunctionModule {
+  /** The exported config function */
+  config?: FunctionConfig;
+  /** The exported getPath function */
+  getPath?: () => string;
+  /** The exported function */
+  default?: ServerlessFunction;
+}
+
+/**
+ * The type definition for serverless function itself.
+ * @public
+ */
+export type ServerlessFunction = (
+  arg: FunctionArgument
+) => HttpFunctionReturnValue;
+
+/**
+ * The exported `config` function's definition.
+ * @public
+ */
+export interface FunctionConfig {
+  /** The name of the serverless function. */
+  name?: string;
+}
+
+/**
+ * The return value for a http/api serverless function
+ * @public
+ */
+export interface HttpFunctionReturnValue {
+  /** HTTP response body (refer to MDN Web Docs). */
+  body: string;
+
+  /** HTTP response status code (refer to MDN Web Docs). */
+  statusCode: number;
+
+  /** HTTP response headers (refer to MDN Web Docs).  */
+  headers: object;
+}
+
+/**
+ * The argument passed to a serverless function or plugin
+ * @public
+ */
+export interface FunctionArgument {
+  /** Object containing each query parameter. */
+  queryParams: { [key: string]: string };
+
+  /** Object containing each path parameter. */
+  pathParams: { [key: string]: string };
+
+  /** Site object containing all deploy-related information. */
+  site: Site;
+}
+
+/**
+ * The return value for an onPageGenerate plugin
+ * @public
+ */
+export interface OnPageGenerateReturnValue {
+  path: string;
+
+  content: string;
+
+  redirects: string[];
+}
+
+/**
+ * The return value for an onUrlChange plugin
+ * @public
+ */
+export type OnUrlChange = HttpFunctionReturnValue;
```

Comment:
yext pages render is fine with onUrlChange plugins returning the same things as http plugins
Not sure if there's supposed to be anything different